Add test for process_add_challenge_message

nsjcorps opened the pull request:
#### Changes proposed in this fix
  - Added test for process_add_challenge_message function

@KhalidRmb , @Ram81 , @RishabhJain2018 , @yashdusing  I have written the test for the process_add_challenge_message function. 

I made one change to the function and that is to raise the error because before the function did not raise the error, it just logged it. Therefore, not allowing the test to run well

My new change works fine now and doesn't break.

Let me know your comments.
Thank you

nsjcorps commented:
@KhalidRmb , @Ram81 , @RishabhJain2018 , @yashdusing . I hope it is fine

pushkalkatara commented:
Hi @nsjcorps Can you please sync the branch with the base branch. The tests look good to me.

nsjcorps commented:
@KhalidRmb , @Ram81 , @RishabhJain2018 , @yashdusing I have done that now.

nsjcorps commented:
@pushkalkatara I have done that now

Ram81 commented on scripts/workers/submission_worker.py:
```diff
@@ -591,6 +591,7 @@ def process_add_challenge_message(message):
         challenge = Challenge.objects.get(id=challenge_id)
     except Challenge.DoesNotExist:
         logger.exception("Challenge {} does not exist".format(challenge_id))
+        raise
```
@nsjcorps I don't think this change should be the part of this PR. It might break things in production. For now let's only add tests in this PR. If we need to raise a exception you can create a separate PR for that

Ram81 replied:
Can you remove this

Ram81 commented on tests/integration/worker/test_submission_worker.py:
```diff
@@ -203,6 +203,29 @@ def test_process_submission_message_when_challenge_phase_does_not_exist(self, mo
+    @mock.patch("scripts.workers.submission_worker.logger.exception")
+    def test_process_add_challenge_message_when_challenge_does_not_exist(self, mock_logger):
+        message = {
+            "challenge_id": self.challenge.pk + 9290,
```
@nsjcorps can't we just change `9290` to `1` ?

nsjcorps commented:
@KhalidRmb , @Ram81 , @RishabhJain2018 , @yashdusing , @pushkalkatara I have updated the code.

I removed my old task because I accidentally left it there

nsjcorps commented:
I have restored all files aside from the task to their original state as shown in the commit.

nsjcorps commented:
@KhalidRmb , @Ram81 , @RishabhJain2018 , @yashdusing , @pushkalkatara I have updated the code.

nsjcorps commented:
In the first commit I accidentally removed the wrong raise statement so I restored that in the second commit and removed the correct one

pushkalkatara commented:
Looks good to me. Please review @Ram81 @yashdusing @RishabhJain2018 @KhalidRmb

nsjcorps commented:
Thanks @pushkalkatara , Any chance I can get this task accepted

yashdusing commented:
@nsjcorps https://codein.withgoogle.com/dashboard/task-instances/5095629450117120/ wasn't the task for the tests of load_challenge function ?

nsjcorps commented:
@yashdusing ,there is another task for the load_challenge function https://codein.withgoogle.com/tasks/5647108686741504/?sp-organization=5932086494494720&sp-is_beginner=False&sp-search=load . The title of this task indicates that it is for the process_add_challenge_message function.

nsjcorps commented:
The load_challenge function is a different task as shown in the link posted above

Ram81 commented on tests/integration/worker/test_submission_worker.py:
```diff
@@ -203,6 +203,29 @@ def test_process_submission_message_when_challenge_phase_does_not_exist(self, mo
 
 
+class ProcessAddChallengeMessageTestClass(BaseTestClass):
+    def test_process_add_challenge_message_successfully(self):
```
@nsjcorps one minor change. Can you please rename this to `test_process_add_challenge_message`

nsjcorps replied:
I have made the requested changes

nsjcorps replied:
@Ram81

nikochiko commented:
> 
> 
> @yashdusing ,there is another task for the load_challenge function https://codein.withgoogle.com/tasks/5647108686741504/?sp-organization=5932086494494720&sp-is_beginner=False&sp-search=load . The title of this task indicates that it is for the process_add_challenge_message function.

@nsjcorps @Ram81 @yashdusing I would like to point out that the task mentioned here is for the tests for `remote_submission_worker` and not for `submission_worker`. So adding tests for `load_challenge` in `submission_worker` should be another task. It would be better if the titles/descriptions of the tasks are corrected. If that is not possible, can we have a different spreadsheet/txt/markdown file with all corrections?
Thanks!

P.S.: I can help with creating the corrections file if required.

nsjcorps commented:
@KhalidRmb , @Ram81 , @RishabhJain2018 , @yashdusing , @pushkalkatara please review

nsjcorps commented:
@yashdusing ,The only property that is needed in the function is the challenge_id. If you look in the process_add_challenge_message function you will see it only gets the challenge_id.

I tested all the possible test cases already. It won't break for any input

nsjcorps commented:
I removed them now seeing as they are unnecessary

Ram81 commented:
@nsjcorps PR LGTM 👍 . Once the issue with `web/test_views.py` we can merge this.
